src/base/datetime/calendar/header/index.ts

```diff
@@ -0,0 +1,295 @@
+  private _yearSelectEl: HTMLSelectElement | undefined;
+
+  @query(".kuc-base-datetime-calendar-header__group__month")
+  private _monthSelectEl: HTMLSelectElement | undefined;
+
+  constructor(props?: BaseDateTimeCalendarHeaderProps) {
+    super();
+    if (!props) {
+      return;
+    }
+    this.language =
+      props.language !== undefined ? props.language : this.language;
+    this.month = props.month !== undefined ? props.month : this.month;
+    this.year = props.year !== undefined ? props.year : this.year;
+  }
+
+  private _handleClickCalendarHeaderButtonPreviousMonth(event: MouseEvent) {
+    event.stopPropagation();
+    if (!this._monthSelectEl || !this._yearSelectEl) return;
+    const monthSelectedIndex = this._monthSelectEl.selectedIndex;
+    const yearSelectedIndex = this._yearSelectEl.selectedIndex;
+    if (monthSelectedIndex === 0) {
+      this._monthSelectEl.selectedIndex = 11;
+      this._yearSelectEl.selectedIndex = yearSelectedIndex - 1;
+    } else {
+      this._monthSelectEl.selectedIndex = monthSelectedIndex - 1;
+    }
+    this._handleChangeCalendarHeader();
+  }
+
+  private _handleClickCalendarHeaderButtonNextMonth(event: MouseEvent) {
+    event.stopPropagation();
+    if (!this._monthSelectEl || !this._yearSelectEl) return;
+    const monthSelectedIndex = this._monthSelectEl.selectedIndex;
+    const yearSelectedIndex = this._yearSelectEl.selectedIndex;
+    if (monthSelectedIndex === 11) {
+      this._monthSelectEl.selectedIndex = 0;
+      this._yearSelectEl.selectedIndex = yearSelectedIndex + 1;
+    } else {
+      this._monthSelectEl.selectedIndex = monthSelectedIndex + 1;
+    }
+    this._handleChangeCalendarHeader();
+  }
+
+  private _handleChangeCalendarHeaderYearSelect(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+    this._handleChangeCalendarHeader();
+  }
+
+  private _handleChangeCalendarHeaderMonthSelect(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+    this._handleChangeCalendarHeader();
+  }
+
+  private _handleChangeCalendarHeader() {
+    const year = this._yearSelectEl?.value;
+    const month = this._monthSelectEl?.value;
+    const detail: CustomEventDetail = { value: `${year}-${month}` };
+    dispatchCustomEvent(this, "kuc:calendar-header-change", detail);
+  }
+
+  private _getYearSelectOptions() {
+    const yearSelectOptions = [];
+    if (!Number.isInteger(this.year)) {
+      this.year = 2021;
+    }
+    for (let i = this.year - 100; i <= this.year + 100; i++) {
+      yearSelectOptions.push(i);
+    }
+    return yearSelectOptions;
+  }
+
+  private _getMonthSelectOptions() {
+    switch (this.language) {
+      case "en":
+        return en.MONTHS_SELECT;
+      case "zh":
+        return zh.MONTHS_SELECT;
+      case "ja":
+        return ja.MONTHS_SELECT;
+      default:
+        return en.MONTHS_SELECT;
+    }
+  }
+
+  private _getYearSelectPostfix() {
```

Comment:
@dong0 
This is dupplication switch/case logic for get locale.
I think we can unify it.
Please considering get locale when an element update.
Ref: https://lit-element.polymer-project.org/guide/lifecycle#update
Ex:
```
export class BaseDateTimeCalendarHeader extends KucBase {
  @property({ type: String }) language = "en";
  
  private locale;
  
  private _getLocale(language) {
    // switch/case
  }
  
  update(changedProperties) {
    // check if "language" changed, set the value to "locale"
    // Ex: this.locale = _getLocale(this.language);
  }
}
```